### PR TITLE
feat: Allow [`identifier`][], understood as [`identifier`][identifier]

### PR DESCRIPTION
**Pull request by oprypin**

### CODE REVIEW

File: tests/test_references.py [hunk 1]
```diff
@@ -121,18 +130,27 @@ def test_missing_reference_with_markdown_text():
...
 def test_missing_reference_with_markdown_implicit():
     """Check that implicit references are not fixed when the identifier is not the exact one."""
     run_references_test(
-        url_map={"Foo": "foo.html#Foo"},
-        source="[`Foo`][]",
-        output="<p>[<code>Foo</code>][]</p>",
-        unmapped=[],
+        url_map={"Foo-bar": "foo.html#Foo-bar"},
+        source="[*Foo-bar*][] and [`Foo`-bar][]",
+        output="<p>[<em>Foo-bar</em>][*Foo-bar*] and [<code>Foo</code>-bar][]</p>",
```
Comment (oprypin) [inline tests/test_references.py:145]: This first part doesn't look great, but it was already like that

File: tests/test_references.py [hunk 2]
```diff
@@ -66,7 +66,16 @@ def test_reference_explicit_with_markdown_text():
...
+        source="This [**Foo**][Foo].",
+        output='<p>This <a href="foo.html#Foo"><strong>Foo</strong></a>.</p>',
+    )
+
+
+def test_reference_implicit_with_code():
+    """Check implicit references (identifier only, wrapped in backticks)."""
+    run_references_test(
+        url_map={"Foo": "foo.html#Foo"},
+        source="This [`Foo`][].",
```
Comment (oprypin) [inline tests/test_references.py:78]: This is the only test that has a different outcome

**Review by oprypin — COMMENTED** [review]
Adding a comment to clarify this because most tests I'm adding are not for the new behavior

File: tests/test_references.py (hunk 1 above)
Comment (pawamoy) [reply tests/test_references.py:145]: You mean `[*Foo-bar*][]` was already transformed as `[<em>Foo-bar</em>][*Foo-bar*]` by the current code?

Comment (oprypin) [reply tests/test_references.py:145]: Yes

Comment (pawamoy) [reply tests/test_references.py:145]: As agreed, we'll deal with this in another issue/PR.

**Comment by pawamoy** [comment]
OK, I feel like it's a nice improvement, and a good base for further improvements. Lets merge.

**Review by pawamoy — APPROVED** [review]
LGTM

**Pull request merged**